### ISSUE TITLE
TFS Team Room Integration

### ISSUE THREAD

**Issue by griffithpsu** [JabbR/JabbR#1164]
We started using Jabbr because we liked the TFS team room, but the user experience stinks.  The reason we like the team room is that we can link defects we're working on into the conversation and maintain that history, which we cant get in Jabbr right now.  We can obviously sync data across from the Jabbr database using the TFS Online API, but that wouldnt be as sexy as a proper TFS plugin for Jabbr.

**Comment by davidfowl** [comment]
I'm a bit confused. You want to use jabbr with TFS team rooms somehow?

**Comment by hnrkndrssn** [comment]
I think what they want is to have TFS notifications pushed to JabbR, like you guys do with GitHub notifications in the SignalR team room. Although I may be **totally** wrong...

**Comment by griffithpsu** [comment]
Yup...thats basically what I'm talking about.  There are really three components of it:

1- TFS currently pushes notifications to its team room (ex. Build Notifications).  It would be cool to see those also pushed to our Jabbr room.  I think we can probably figure this out on our end, as TFS online provides service hooks for these notifications (ex. They provide a hook that will push data to Trello).  They obviously dont have one out of the box for Jabbr, but they do have one for Azure Service Bus...so we could tap into that and then build something that wraps a Jabbr API.
http://blogs.msdn.com/b/somasegar/archive/2014/05/12/visual-studio-online-rest-apis-and-service-hooks.aspx

2 - Right now in TFS you can easily navigate conversation history that took place in the TFS Team room and use @ and # characters to associate conversation to individuals or TFS work items (i.e. stories, defects, etc).  So it is beneficial to have conversation data stored in TFS, but we prefer working in Jabbr.  So the ability to have Jabbr push the conversation from a room into TFS so that we can leverage the TFS history and conversation association features.
http://msdn.microsoft.com/en-us/library/dn169471.aspx

Note...we could also do this by tapping into the Jabbr DB and using the TFS API, but an 'in-box' solution would be cool.

3 - Finally, as noted in item 2 above, in the TFS team room if you use an # and the work item number it is clickable directly to the defect, backlog item, etc.  This is more of a nice to have feature, but it would be cool to enable that linking with TFS.

Btw - Great work on Jabbr guys.....our team has been loving it.  Keep up the great work!
